src/recipe/parser/build.rs

```diff
@@ -238,8 +238,12 @@ impl Build {
     }
 
     /// Get the noarch type.
-    pub const fn noarch(&self) -> &NoArchType {
-        &self.noarch
+    pub fn noarch(&self) -> NoArchType {
+        if self.python().version_independent {
+            return NoArchType::python();
```

Comment:
Why do this?

In conda-build PR what I did was to check `python_version_independent` which was the union of `python_version_independent` setting and `noarch: python` setting.